here-naksha-lib-psql/src/test/java/com/here/naksha/lib/psql/PsqlStorageTests.java

```diff
@@ -404,6 +404,70 @@ void singleFeatureUpdateVerify() throws NoCursor {
     }
   }
 
+  @Test
+  @Order(56)
+  @EnabledIf("runTest")
+  void singleFeatureGetAllVersions() throws NoCursor {
+    assertNotNull(storage);
+    assertNotNull(session);
+    // given
+    /**
+     * data inserted in {@link #singleFeatureCreate()} test and updated by {@link #singleFeatureUpdate()}.
+     */
+    final ReadFeatures request = RequestHelper.readFeaturesByIdRequest(collectionId(), SINGLE_FEATURE_ID);
+    request.withReturnAllVersions(true);
+
+    // when
+    try (final MutableCursor<XyzFeature, XyzFeatureCodec> cursor =
+             session.execute(request).getXyzMutableCursor()) {
+      cursor.next();
+      XyzFeature ver1 = cursor.getFeature();
+      cursor.next();
+      XyzFeature ver2 = cursor.getFeature();
+      cursor.next();
+      XyzFeature ver3 = cursor.getFeature();
+      assertFalse(cursor.hasNext());
+      assertEquals(ver1.getId(), ver2.getId());
+      assertEquals(ver1.getId(), ver3.getId());
+      assertNotEquals(ver1.getProperties().getXyzNamespace().getTxn(), ver2.getProperties().getXyzNamespace().getTxn());
+    }
+  }
+
+  @Test
+  @Order(56)
+  @EnabledIf("runTest")
+  void singleFeatureGetSpecificVersionRead() throws NoCursor {
```

Comment:
I am trying to understand how we can achieve this requirement:
`support fetching documents for given versions (UUIDs), to be able to find feature diff between in-hand and its previous version (obtained using UUID).`

Assume, we fetched following records from the history (let's based on some property search):
1. Feature 1 - v5
2. Feature 1 - v4
3. Feature 2 - v10
4. Feature 2 - v9

Now, question is - how to retrieve **previous** version of 2nd record ? (i.e. previous version of `Feature 1 - v4`).

In my view, we should be able to do this by querying `uuid` = `puuid` of `Feature 1 - v4`

Is that correct understanding? Or may be some other better way?

In either case - I would like to see such test case.

Reply:
yes `uuid` = `puuid`, is exactly what you're looking for

Reply:
I added section in test that covers it:
`// get previous version by uuid = puuid`